### DISCUSSION TITLE
How to debug custom Python script executed by mitmdump in vscode?

### DISCUSSION THREAD

**Discussion by zone-coba** [mitmproxy/mitmproxy#7318] [General]
I run this command

`mitmdump -s proxy.py -p 80 &`

from a CMD in a Docker container. I need to debug my custom Python script _proxy.py_ in **vscode**.

Status:
- I'm able to get the container running with ports exposed and the debugger attached in vscode. All looks good.
- I can curl the service and get a response - but no breakpoint in my code is hit. Somehow debugpy isn't able to hook on my custom code.

Most probably whis is due to the way the Python code is executed by _mitmdump_ - as a native command which then executes my code, and not directly using python3 runtime. 

Tried so many things, but now ran out of ideas. 
Any ideas? Any hints appreciated!

**Comment by mhils** [comment]
Debugging add-ons with PyCharm breakpoints (no Docker) works fine for me,
but that's the only data point I have. Have you tried to just call
breakpoint() in your code?

**Reply by zone-coba** [reply]
Good idea, tried this, but adding breakpoint() somewhere in request(flow: http.HTTPFlow) breaks it with Addon error and I get 503 when curling it.

**Reply by zone-coba** [reply]
I wonder how I could convert the command `mitmdump -s proxy.py -p 80 &` into a python invocation but didn't find a way yet. Just running it like `subprocess.run(["mitmdump", "-s", "proxy/proxy.py", "-p", "80"]) `doesn't do the trick, debugger still doesn't attach to the custom code.
